gh-131670: Fix crash in `anext()` when `__anext__` is sync and raises (#131682)
python/cpython: Misc/NEWS.d/next/Core_and_Builtins/2025-03-24-19-38-53.gh-issue-131670.IffOZj.rst

> Fix :func:`anext` failing on sync :meth:`~object.__anext__` that raises
> :exc:`StopAsyncIteration`.

Reviewer: ```suggestion
Fix :func:`anext` failing on sync :meth:`~object.__anext__` raising an exception.
```
